scripts/pre_commit_linter.py

```diff
@@ -80,34 +84,67 @@ def _is_javascript_file(filename):
     return filename.endswith('.js')
 
 
-def _lint_js_files(node_path, jscs_path, autofix, config_jscsrc):
+def _get_all_files_in_directory(dir_path):
+    """Recursively collects all files in directory and 
+    subdirectories of specified path.
+
+    Args:
+    - dir_path: str. Path to the folder to be linted.
+
+    Returns:
+        a list of files in directory and subdirectories.
+    """
+    for subdirectory in os.listdir(dir_path):
```

Comment:
FYI you could also use os.walk -- see scripts/build.py. Then you don't need the files_in_directory global.